server/grpc_service.go

```diff
@@ -1990,7 +1991,12 @@ func (s *GrpcServer) WatchGlobalConfig(req *pdpb.WatchGlobalConfigRequest, serve
 			for _, e := range res.Events {
 				// Since item value needs to support marshal of different struct types,
 				// it should be set to `Payload bytes` instead of `Value string`.
-				cfgs = append(cfgs, &pdpb.GlobalConfigItem{Name: string(e.Kv.Key), Payload: e.Kv.Value, Kind: pdpb.EventType(e.Type)})
+				switch e.Type {
+				case clientv3.EventTypePut:
+					cfgs = append(cfgs, &pdpb.GlobalConfigItem{Name: string(e.Kv.Key), Payload: e.Kv.Value, Kind: pdpb.EventType(e.Type)})
+				case clientv3.EventTypeDelete:
+					cfgs = append(cfgs, &pdpb.GlobalConfigItem{Name: string(e.Kv.Key), Payload: e.PrevKv.Value, Kind: pdpb.EventType(e.Type)})
```

Comment:
According to the document of `WithPrevKV`:

> WithPrevKV gets the previous key-value pair before the event happens. If the previous KV is already compacted, nothing will be returned.

When the previous KV is already compacted, will `e.PrevKv.Value` panic? If not, what should we do to handle the empty value?

Reply:
If the previous KV is compacted, it should not be a delete event.

Reply:
‘the previous KV is already compacted’ can only happen when delete is called multiple times, but this should not be our case.

Reply:
If the previous KV is already compacted, e.PrevKv will be nil, so accessing its fields will not panic, just judge for nil
do u have any idea?